sourced/ml/core/algorithms/token_parser.py

```diff
@@ -49,9 +54,17 @@ def __init__(self, stem_threshold=STEM_THRESHOLD, max_token_length=MAX_TOKEN_LEN
         self._single_shot = single_shot
         self._save_token_style = save_token_style
         self._attach_upper = attach_upper
+        self._nn_model = nn_model
+        self._id_splitter_nn = None
+        if use_nn or self._nn_model is not None:
```

Comment:
```suggestion
        if use_nn:
```
It is fine for it to be None.

Reply:
Yup totally, I just assumed if one specify a NN model that means he wants to use the nn model parser WDYT ?

Reply:
If we set `use_nn=True, nn_model=None` then we want the most up-to-date model from the registry. `Model.load(None)` should work perfectly.